src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -214,7 +223,9 @@ private void executeQuery(BigQuery bigQuery, QueryJobConfiguration queryConfig,
       if (RETRY_ON_REASON.contains(queryJob.getStatus().getError().getReason())) {
         throw new BigQueryJobExecutionException(queryJob.getStatus().getError().getMessage());
```

Comment:
change it to `ProgramFailureException`, this comment applies to whole file wherever this exception is thrown.